packages/x-tree-view-pro/src/internals/plugins/useTreeViewItemsReordering/useTreeViewItemsReordering.ts

```diff
@@ -84,7 +84,7 @@ export const useTreeViewItemsReordering: TreeViewPlugin<UseTreeViewItemsReorderi
           isValid = false;
         } else if (canMoveItemToNewPosition) {
           isValid = canMoveItemToNewPosition({
-            itemId,
+            itemId: state.itemsReordering!.draggedItemId,
```

Comment:
shouldn't this be picked up by TS automatically? I mean there is an `if` condition that would essentially discriminate the possibility of it being `undefined`/`null`

Reply:
It is not picked up by TS automatically because we are inside a function `checkIfPositionIsValid`, so TS considers that we might have mutated this object between the `if` line 62 and the line your commented :cry:

Reply:
yeah ... i thought so ... weird shortcoming of TS in this case 👍🏼

Reply:
`state.itemsReordering` is perceived as "mutable", so `typescript` can't ensure it is the same between the two usages. To overcome that you can move the related var into a `const`

```tsx
const itemsReordering = state.itemsReordering
...
if (itemsReordering)
  myFuncWithRequiredProp(itemsReordering)
```

Reply:
You can also define the type that defines the object as `readonly` 😅

Reply:
I tried the `reaonly` on both `state` and `state.itemsReordering` an TS still complains :cry:

Reply:
Probably because the param os the function is not readonly, then `state` itself can be another instance of `state`? I'm unsure 🤔  

Anyways, I wouldn't dive into this rabbit hole just for this, as there are a lot of conditions for that to work correctly. The other alternative (using a `const`) should work properly

Reply:
Done